sql/src/main/java/io/crate/operation/auth/AuthenticationProvider.java

```diff
@@ -77,7 +95,7 @@ public boolean enabled() {
         }
 
         @Override
-        public AuthenticationMethod resolveAuthenticationType(String user, InetAddress address) {
+        public AuthenticationMethod resolveAuthenticationType(String user, InetAddress address, String protocol) {
```

Comment:
Couldn't we change the method signature to take the enum / `set<enum>` as well - this would have the benefit that the type system would also document the allowed inputs + compile check them.